source/common/upstream/eds.cc

```diff
@@ -251,8 +251,12 @@ bool EdsClusterImpl::updateHostsPerLocality(
   return false;
 }
 
-void EdsClusterImpl::onConfigUpdateFailed(const EnvoyException* e) {
-  UNREFERENCED_PARAMETER(e);
+void EdsClusterImpl::onConfigUpdateFailed(Envoy::Config::ConfigUpdateFailureReason,
+                                          const EnvoyException* e) {
+  //  We should not call onPreInitComplete if this method called because of stream disconnection.
+  if (e == nullptr) {
```

Comment:
Why not use the ConfigUpdateFailureReason here rather than the indirect `e == nullptr`? This seems to be one of the few places (and motivating example) there is benefit from plumbing this reason in to the subscriber callbacks, but we 're not using it.

Reply:
Great Catch @htuch. I meant to change this and forgot. Thanks. Changed now, PTAL.